### PR TITLE
APP-6696 platform tagging for pi

### PR DESCRIPTION
**Pull request by abe-winter**
## What changed
- add `pi` and `pifull` tags to the extended platform tags
- look at the new test to see what the outputs will look like, but basically it will be `[pi:4, pifull:cm4S]`
## Why
For targeting if needed.
## Pre-merge checklist
- [x] once approved I'll test on a device

Logging output on device looks good:
> 2024-11-05T17:38:35.426Z	INFO	rdk	server/entrypoint.go:69	Viam RDK	{"git_rev":"9ba7b7b5e9260d72b03498c571f3d38bc0b19098"}

> 2024-11-05T17:38:36.358Z	INFO	rdk	config/platform.go:157	platform tags	{"tags":"distro:debian,os_version:12,codename:bookworm,**pi:5,pifull:5B**"}

### CODE REVIEW

**Review by JohnN193 — APPROVED** [review]

File: config/platform.go [hunk 1]
```diff
@@ -47,6 +48,56 @@ func readGPUTags(logger logging.Logger, tags []string) []string {
...
+			ret.version = "1"
+		}
+		ret.longVersion += ret.version
+		ret.version = ret.version[:1] // contract 3E to 3 now that it's been copied to longVersion
+		switch litePlus {
+		case "Lite":
+			ret.longVersion += "l"
+		case "Plus":
+			ret.longVersion += "p"
+		}
```
Comment (randhid) [inline config/platform.go:78]: Is there a default case we haven't hit in all our pis?

**Review by randhid — APPROVED** [review]
Honeslty lgtm.

File: config/platform.go (hunk 1 above)
Comment (abe-winter) [reply config/platform.go:78]: you mean should this switch stmt have a default?

Comment (randhid) [reply config/platform.go:78]: yep to *

Comment (abe-winter) [reply config/platform.go]: added warning in bd228f1b3

**Pull request merged**